plugin.video.embycon/resources/lib/simple_logging.py

```diff
@@ -0,0 +1,46 @@
+# Gnu General Public License - see LICENSE.TXT
+
+import xbmc
+import xbmcaddon
+
+
+class SimpleLogging():
+    level = 0
+    name = ""
+
+    def __init__(self, name):
+        settings = xbmcaddon.Addon(id='plugin.video.embycon')
+        prefix = settings.getAddonInfo('name')
+        log_level = settings.getSetting('logLevel')
+        self.level = int(log_level)
+        self.name = prefix + '.' + name
+
+    def getLevel(self):
+        return self.level
+
+    def __str__(self):
+        return "LogLevel: " + str(self.level)
+
+    def error(self, msg):
+        if (self.level >= 0):
+            try:
+                xbmc.log(self.format(msg, "ERROR"), level=xbmc.LOGNOTICE)
+            except UnicodeEncodeError:
+                xbmc.log(self.format(msg, "ERROR").encode('utf-8'), level=xbmc.LOGNOTICE)
+
+    def info(self, msg):
+        if (self.level >= 1):
+            try:
+                xbmc.log(self.format(msg, "INFO"), level=xbmc.LOGNOTICE)
+            except UnicodeEncodeError:
+                xbmc.log(self.format(msg, "INFO").encode('utf-8'), level=xbmc.LOGNOTICE)
+
+    def debug(self, msg):
+        if (self.level >= 2):
+            try:
+                xbmc.log(self.format(msg, "DEBUG"), level=xbmc.LOGNOTICE)
```

Comment:
the problem with this is if you want to show addon debugging log info you need to enable debugging logging in two places, the addon and also Kodi, this also prints out all the Kodi debug info which causes logs to be huge and hard to manage for submission.
If this is a problem I guess I can do this but it will make my life harder.